dpnp/dpnp_iface_manipulation.py

```diff
@@ -2120,16 +2124,26 @@ def reshape(a, /, newshape, order="C", copy=None):
 
     """
 
-    if newshape is None:
-        newshape = a.shape
+    if newshape is None and shape is None:
+        raise TypeError(
+            "reshape() missing 1 required positional argument: 'shape'"
+        )
+
+    if newshape is not None:
+        if shape is not None:
+            raise TypeError(
+                "You cannot specify 'newshape' and 'shape' arguments "
+                "at the same time."
+            )
+        shape = newshape
```

Comment:
Then it makes sense to add a deprecation warning here, like in numpy. Since in the future we will have to remove that deprecated behavior anyway.